quizzes/quiz6.md

```diff
@@ -0,0 +1,313 @@
+
+```js
+let a = {
+    amount: 10,
+    increase: function() {
+        this.amount = 20;
+    }
+}
+let b = {
+    amount: 15,
+    increase: function() {
+        this.amount = 20;
+    }
+}
+a.increase(); // "this" variable is referring to variable a
+b.increase(); // "this" variable is referring to variable b
+```
+
+With this new information, we can reimplement our code from before to fix the bug.
+
+```js
+let john = {
+    name: "John",
+    isMarried: false,
+    livesIn: "2 bed room flat",
+    marryTo: function(spName) {
+        this.isMarried = true;
+        this.supposesName = spName;
+        this.livesIn = "house";
+        this.numberOfChildren = 0;
+    }
+}
+
+let adam = john;
+john = null;
+adam.name = "Adam";
+adam.marryTo("Jane");
+```
+
+Now, when the `adam.marryTo("Jane")` method runs, it will safely change the `adam` object.
+
+By the way, there is still a small thing we can improve about our code, but I am not going to bore you with details here. If you want to know, see me in the class and ask me about the code duplication.
+
+---
+
+**document.createElement**
+
+One big part of web development is to start with a minimal amount of html code which is static by nature and add new elements to the web page on the fly. Meaning that, you will make things appear inside the browser screen area but as soon as the browser tab closes, the dynamic content will disappear.
+
+Your number one tool for adding new elements to your dom is linked to mastering the document.createElement method. Likely it is very easy to master. All you have to do is to provide the type of element you wish to create as input parameter. To create a `div` element, simply say:
+
+```js
+document.createElement("div");
+```
+
+For an input element, say:
+
+```js
+document.createElement("input");
+```
+
+The real fun begins, when you start modifying the element you created, but for that you need to save the output of `createElement` like this:
+
+```js
+let divElement = document.createElement("div");
+let inputElement = document.createElement("input");
+```
+
+Now we can modify away everything about our elements. For example:
+
+```js
+divElement.textContent = "God day to you sir";
+divElement.style.backgroundColor = "red";
+
+inputElement.type = "number";
+inputElement.onchange = function() {
+    console.log("user entered a new value");
+}
+```
+
+So in short, whenever you need to add new content to your dom, you start with creating a new object. Initially, every object you create using this method will be invisible to the user. You can add this element anywhere in your dom hierarchy. Because of this variability, you need to specify where you want this object to be added. For learning how to do that keep on reading...
```

Comment:
```suggestion
So in short, whenever you need to add new content to your DOM, you start with creating a new object. Initially, every object you create using this method will be invisible to the user. You can add this element anywhere in your DOM hierarchy. Because of this variability, you need to specify where you want this object to be added. For learning how to do that keep on reading...
```